pycqed/instrument_drivers/meta_instrument/qubit_objects/QuDev_transmon.py

```diff
@@ -1290,29 +1274,54 @@ def measure_randomized_benchmarking(
         MC.set_sweep_points_2D(soft_sweep_points)
 
+        data_processing_pipe = []
         if thresholded:
-            MC.set_detector_function(self.dig_avg_det)
+            det_func = self.dig_avg_det
         elif classified_ro:
-            MC.set_detector_function(self.int_avg_classif_det)
+            det_func = self.int_avg_classif_det
         else:
-            MC.set_detector_function(self.int_avg_det)
+            det_func = self.int_avg_det
+        MC.set_detector_function(det_func)
+
+        data_processing_pipe += [
+            # average data
+            {'node_type': 'average',
+             'num_bins': [len(cliffords)]*len(det_func.value_names),
+             'data_keys_in': det_func.value_names,
+             'data_keys_out': [e+'_avg' for e in det_func.value_names], **kw},
+            # std for each clifford sequence
+            {'node_type': 'get_std_deviation',
+             'num_bins': [len(cliffords)]*len(det_func.value_names),
+             'data_keys_in': det_func.value_names,
+             'data_keys_out': [e+'_std' for e in det_func.value_names], **kw},
+            # do SgQbRB
+            {'node_type': 'SingleQubitRBAnalysis',
+             'std_keys': [e+'_std' for e in det_func.value_names],
+             'data_keys_in': [e+'_avg' for e in det_func.value_names],
+             'meas_obj_name': self.name, **kw}]
+
+        # create sweep points
+        sp = SweepPoints('nr_seeds', np.arange(nr_seeds), '', 'Nr. Seeds')
+        sp.add_sweep_dimension()
+        sp.add_sweep_parameter('cliffords', cliffords, '',
+                               'Number of applied Cliffords, $m$')
         if exp_metadata is None:
             exp_metadata = {}
         exp_metadata.update({'preparation_params': prep_params,
                              'cal_points': repr(cp),
-                             'rotate': len(cp.states) != 0,
-                             'data_to_fit': {self.name: 'pe'},
-                             'hard_sweep_params':
-                                 {'nr_seeds': {'values': np.arange(nr_seeds),
-                                               'unit': ''}},
-                             'soft_sweep_params':
-                                 {'cliffords': {'values': cliffords,
-                                                'unit': ''}}})
+                             'sweep_points': sp,
+                             'meas_obj_sweep_points_map':
+                                 sp.get_sweep_points_map([self.name]),
+                             'meas_obj_value_names_map': {
+                                 self.name: det_func.value_names},
+                             'processing_pipe': data_processing_pipe})
         MC.run_2D(label, exp_metadata=exp_metadata)
 
         if analyze:
-            tda.MultiQubit_TimeDomain_Analysis(
-                qb_names=[self.name], options_dict={'TwoD': True})
+            if classified_ro:
+                ba3.BaseDataAnalysis()
```

Comment:
Why do we run the base analysis here? Don't we have an analysis for normal RB?

Reply:
See above comment.